src/Api.js

```diff
@@ -0,0 +1,37 @@
+"use strict";
+
+const _ = require("lodash");
+
+/**
+ * @typedef {object} ApiOptions
+ * @property cookies {object}
+ */
+
+class Api {
+
+  /**
+   *
+   * @param baseUrl {string}
+   * @param options {ApiOptions}
+   */
+  constructor(baseUrl, options) {
+    if (this.constructor === Api) {
+      throw new Error("Abstract class 'Api' cannot be instantiated!");
```

Comment:
[Minor] Wäre hier nicht eine `abstract class` sinnvoll? https://www.typescriptlang.org/docs/handbook/2/classes.html#abstract-classes-and-members
Dann bräuchte man dieses Konstrukt gar nicht, da TypeScript schon verhindert dass man eine abstract class nicht instanzieren kann. Vor allem kommt dieser Fehler erst zur RunTime. Mit der abstract class würde es schon vorher auffallen.

Reply:
@Zwergal Natürlich war das auch mein Gedanken zuerst ;-) Aber wir haben hier noch kein TS.

Reply:
ah ok. hatte nur die types.d.ts gesehen und dachte wir machen hier schon ts :)
Dann passt es.